pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java

```diff
@@ -63,10 +63,18 @@ protected void internalUpdateResourceGroup(String rgName, ResourceGroup rgConfig
             /*
              * assuming read-modify-write
              */
-            resourceGroup.setPublishRateInMsgs(rgConfig.getPublishRateInMsgs());
-            resourceGroup.setPublishRateInBytes(rgConfig.getPublishRateInBytes());
-            resourceGroup.setDispatchRateInMsgs(rgConfig.getDispatchRateInMsgs());
-            resourceGroup.setDispatchRateInBytes(rgConfig.getDispatchRateInBytes());
+            if (rgConfig.getPublishRateInMsgs() != -1) {
```

Comment:
What does `-1` means here? do not have a limit for publish rate? or `0` means no limit for publish rate?

Reply:
The default value of `rate` is `- 1`
https://github.com/apache/pulsar/blob/ef8120f0ae16d829036292dc0b50640bc515ae98/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java#L87-L106
https://github.com/apache/pulsar/blob/ef8120f0ae16d829036292dc0b50640bc515ae98/pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/PublishRate.java#L28-L37
which means not take effect or not limit.

Reply:
@codelipenghui ptal again, thx!

Reply:
Yes, this was why I am considering this part. If the existing resource group is 

```java
private int publishRateInMsgs = 10; 
private long publishRateInBytes = 10; 
private int dispatchRateInMsgs = 10; 
private long dispatchRateInBytes = 10;
```

And want to remove the limit for `dispatchRateInMsgs`, the new resource group should be:

```java
private int publishRateInMsgs = 10; 
private long publishRateInBytes = 10; 
private int dispatchRateInMsgs = -1; 
private long dispatchRateInBytes = 10;
```

But we can't apply `dispatchRateInMsgs = -1` in your fix right?

Reply:
Good point, you are right!
`-1` is an occupied value, we can‘t judge whether `-1` is the default value or  user-specified value, so maybe we should remove the `-1` and judge by `null`, redefine `ResourceGroup`
```
 public class ResourceGroup { 
     private int publishRateInMsgs = -1; 
     private long publishRateInBytes = -1; 
     private int dispatchRateInMsgs = -1; 
     private long dispatchRateInBytes = -1; 
 } 
```